### PR TITLE
Perf/experimental performance features

### PR DESCRIPTION
**Pull request by saadataz**
- Some quick refactoring
- Added Experimental features switches for:
- [x] - disable arc-exclude
- [x] - focus only elements with tabIndex

- Benchmarks added!
image

### CODE REVIEW

File: src/focus.service.ts [hunk 1]
```diff
@@ -588,43 +598,49 @@ export class FocusService {
...
-      return false;
-    }
+    if (!this.isVisible(el)) { return false; }
+
+    if (this.focusTabIndexOnly) { return true; }
 
-    for (let parent = el; parent; parent = parent.parentElement) {
-      const parentRecord = this.registry.find(parent);
-      if (parentRecord && parentRecord.exclude && parentRecord.exclude()) {
+    // Eventually depricate arc-exclude as it is rarely used but consumes CPU cycles
```
Comment (connor4312) [inline src/focus.service.ts]: Can we benchmark this? I very much doubt this is a bottleneck, transversing up the tree should run at many millions of ops/sec

File: demo/webpack.config.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ module.exports = {
   },
   devServer: {
     host: '0.0.0.0',
-    port: 8080,
+    port: 8087,
```
Comment (SimonSchick) [inline demo/webpack.config.js]: undo

File: src/focus.service.ts [hunk 3]
```diff
@@ -671,9 +687,11 @@ export class FocusService {
     // method of transversal would be slow, but it's actually really freaking
     // fast. Like, 6 million op/sec on complex pages. So don't bother trying
     // to optimize it unless you have to.
-    const focusableElems = this.focusRoot.querySelectorAll('*');
+    const focusableElems = this.focusTabIndexOnly ? this.focusRoot.querySelectorAll('[tabindex') : this.focusRoot.querySelectorAll('*');
```
Comment (connor4312) [inline src/focus.service.ts]: `'[tabindex'` is invalid

File: src/focus.service.ts (hunk 1 above)
Comment (saadataz) [inline src/focus.service.ts]: Its not just parent lookup but also registry.find thats consuming resources. On top of that, code runs of every potential element.
Let me see how I can benchmark this

Comment (connor4312) [inline src/focus.service.ts]: Getting a solid bench setup, perhaps in the demo module, would be superb

**Pull request merged**